include/swift/AST/DiagnosticsSema.def

```diff
@@ -1317,20 +1328,25 @@ ERROR(unsupported_nested_protocol,none,
 // Type aliases
 ERROR(type_alias_underlying_type_access,none,
       "type alias %select{must be declared %select{"
-      "%select{private|fileprivate|internal|%error|%error}2|private or fileprivate}3"
+      "%select{private|fileprivate|internal|%error|%error}1|private or fileprivate}3"
```

Comment:
(here and below) This was correct the way it was before. Argument 1 is the current access; argument 2 is the required access.

Reply:
Are you sure? When isExplicit is true, %1 is the decl's getFormalAccess(), when isExplicit is false, %1 is the AccessScope's `requiredAccessForDiagnostics()`. On the other hand, %2 is the AccessScope's `accessLevelForDiagnostics()`.

I noticed some inconsistency here and cleaned it up. Some diagnostics were using %2 here for the isExplicit == false case, but this appeared to generate a wrong diagnostic because of the distinction between `requiredAccessForDiagnostics()` and `accessLevelForDiagnostics()` with private vs fileprivate. This change and tests for it was in a separate patch so it should be easier to review that way.

Reply:
Note the difference in messages: "must be declared X" vs. "cannot be declared Y". There might be mistakes at some of the use sites, but they're definitely not supposed to be the same value.

Reply:
"When isExplicit is true, %1 is the decl's getFormalAccess(), when isExplicit is false, %1 is the AccessScope's requiredAccessForDiagnostics()" -- so they're different values :)

So using %1 vs %2 in the !isExplicit case is almost equivalent, except for private vs fileprivate.

Reply:
I see: you changed the call site too. Please don't do this; it muddles the notion of what each parameter is used for. At that point I'd rather just use different diagnostics.